pkg/agent/cniserver/ipam/antrea_ipam.go

```diff
@@ -153,23 +159,20 @@ func (d *AntreaIPAM) Add(args *invoke.Args, k8sArgs *argtypes.K8sArgs, networkCo
 }
 
 // Del deletes IP associated with resource from IP Pool status
-func (d *AntreaIPAM) Del(args *invoke.Args, k8sArgs *argtypes.K8sArgs, networkConfig []byte) (bool, error) {
-	mine, allocator, _, _, err := d.owns(k8sArgs)
-	if mine == mineFalse || mine == mineUnknown {
-		// pass this request to next driver
-		return false, nil
-	}
+func (d *AntreaIPAM) Del(args *invoke.Args, k8sArgs *types.K8sArgs, networkConfig []byte) (bool, error) {
+	owner := getAllocationOwner(args, k8sArgs, nil, false)
+	foundAllocation, err := d.del(owner.Pod)
 	if err != nil {
+		// Let the invoker retry at error.
 		return true, err
 	}
 
-	owner := getAllocationOwner(args, k8sArgs, nil)
-	err = allocator.ReleaseContainerIfPresent(owner.Pod.ContainerID)
-	return true, err
+	// If no allocation found, pass CNI DEL to the next driver.
+	return foundAllocation, nil
```

Comment:
Here if allocation was not found (due to duplicate del, for example) we might be passing irrelevant request to next driver. I see no functional harm in it but it might be confusing.

Reply:
Yes. But we have no way to know we indeed should pass to the next driver or not right? @gran-vmv: as you implemented the behavior, you might comment have you verified any issue if the next driver (host-local) does not handle the allocation? I assume it will just silently return no error.

But maybe we can add more comments here to explain the situation.

Reply:
The previous implementation was more robust since it mainly relied on annotation rather than pool status. But I tend to think that code simplicity that resulted from your refactoring is worth this cost.

Reply:
But on the another aspect - it is not safe to rely on the Pod annotation either as it can be changed. And even the previous implementation can still get the same when the Pod resource is already deleted.